framework_tvb/tvb/core/services/burst_service.py

```diff
@@ -102,6 +103,11 @@ def persist_operation_state(self, operation, operation_status, message=None):
         dao.store_entity(operation)
         operation = dao.get_operation_by_id(operation.id)
         self.file_helper.write_operation_metadata(operation)
+        #update burst also
+        burst_id = json.loads(operation.meta_data)[DataTypeMetaData.KEY_BURST]
+        if burst_id:
+            burst_config = self.load_burst_configuration(burst_id)
+            self.mark_burst_finished(burst_config, burst_config.BURST_ERROR, message)
```

Comment:
This can not be right for the following reasons:
- it should be `burst_config = self.get_burst_for_operation_id(operation.id)` (instead of parsing the `operation.meta_data`)
- not always an operation has a linked burst, so a check for `if burst_config is None:` afterwards will be necessary
- in this method we receive `operation_status`, (could be OK, ERROR, CANCELED) while in the new code it is always put `burst_config.BURST_ERROR`
- I see places, like the `backend_client.py` were `burst_service.persist_operation_state` and `burst_service.mark_burst_finished` are called consecutively. With the new code the BurstConfig will be unnecessarily updated twice